prime-dotnet-webapi/Controllers/OrganizationsController.cs

```diff
@@ -110,6 +123,116 @@ public async Task<ActionResult<Organization>> CreateOrganization(CreateOrganizat
             );
         }
 
+        // POST: api/Organizations/claim
+        /// <summary>
+        /// Claim an existing Organization.
+        /// </summary>
+        [HttpPost("claim", Name = nameof(ClaimOrganization))]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ApiResultResponse<int>), StatusCodes.Status200OK)]
+
+        public async Task<ActionResult<Organization>> ClaimOrganization(ClaimOrganizationViewModel claimOrganization)
+        {
+            if (!await _partyService.PartyExistsAsync(claimOrganization.PartyId, PartyType.SigningAuthority))
+            {
+                return BadRequest("Could not claim an organization, the passed in SigningAuthority does not exist.");
+            }
+
+            var organization = await _organizationService.ClaimOrganizationAsync(claimOrganization.PartyId, claimOrganization.PEC, claimOrganization.ClaimDetail);
+
+            return Ok(organization);
```

Comment:
I _very strongly_ feel this endpoint should not be returning the whole `Organization` back to the user. The annotation for this endpoint reads `[ProducesResponseType(typeof(ApiResultResponse<int>), StatusCodes.Status200OK)]`, was the original intention for this endpoint to return just the OrganizationId of the org? What does the FE actually require?

Reply:
It is true that organization returned is not being used if it is a claim. This was a quick workaround after discussion with @mtpultz. The FE expects the Organization object to be returned to satisfy the `afterSubmitIsSuccessful` method of the OrganizationSigningAuthorityPage (inherits from `AbstractEnrolmentPage`).

> I _very strongly_ feel this endpoint should not be returning the whole `Organization` back to the user. The annotation for this endpoint reads `[ProducesResponseType(typeof(ApiResultResponse<int>), StatusCodes.Status200OK)]`, was the original intention for this endpoint to return just the OrganizationId of the org? What does the FE actually require?